module/bdev/raid/bdev_raid.c

```diff
@@ -3367,10 +3870,17 @@ raid_bdev_add_base_bdev(struct raid_bdev *raid_bdev, const char *name,
 
 	if (base_info == NULL || raid_bdev->state == RAID_BDEV_STATE_ONLINE) {
 		RAID_FOR_EACH_BASE_BDEV(raid_bdev, iter) {
-			if (iter->name == NULL && spdk_uuid_is_null(&iter->uuid)) {
+			if (iter->name == NULL && spdk_uuid_is_null(&iter->uuid) && iter->state == BASE_BDEV_STATE_NONE) {
 				base_info = iter;
 				break;
 			}
+
+			/* Check if a faulty base bdev with the same name is present */
+			if (iter->state != BASE_BDEV_STATE_NONE && strcmp(iter->name, name) == 0) {
```

Comment:
Using`strncmp` is safer

Reply:
Yes it is, if you know at least the length of one of the 2 strings to be compared. We don't have such an information, so we can't use it.